### PR TITLE
Validate URLs in web commands before execution

### PR DESCRIPTION
**Pull request by edcohen08**
### Background

This pr targets  #1887 to validate urls before browsing. It was implemented in web_requests get_response but not for selenium or git.

### Changes

Adds a validator for urls that just builds off of the validation that existed in get_response and adds it as a wrapper to add to commands that have a url as the first param.

### Documentation

Documented in code

### Test Plan

Still passes test suite and tested locally

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/commands/web_playwright.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from bs4 import BeautifulSoup
 
 from autogpt.processing.html import extract_hyperlinks, format_hyperlinks
+from autogpt.url_utils.validators import validate_url
```
Comment (hdkiller) [inline autogpt/commands/web_playwright.py]: I did not found the decorator being used within playwright.py

File: autogpt/url_utils/validators.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import functools
+from typing import Any, Callable
+from urllib.parse import urljoin, urlparse
+
+from requests.compat import urljoin
```
Comment (hdkiller) [inline autogpt/url_utils/validators.py:5]: Missing Returns / Raises

File: autogpt/url_utils/validators.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def validate_url(func: Callable[..., Any]) -> Any:
+    """The command validate_url is used to validate urls for any command that requires
+    a url as an arugment"""
+
```
Comment (hdkiller) [inline autogpt/url_utils/validators.py:11]: Args / Returns / Raises also missing from here.

**Review by hdkiller — DISMISSED** [review]
I've also tested this PR and it's works for me.

File: autogpt/url_utils/validators.py (hunk 3 above)
Comment (edcohen08) [reply autogpt/url_utils/validators.py:11]: nice catch!

File: autogpt/url_utils/validators.py (hunk 2 above)
Comment (edcohen08) [reply autogpt/url_utils/validators.py:5]: ah thank ya

File: autogpt/commands/web_playwright.py (hunk 1 above)
Comment (edcohen08) [reply autogpt/commands/web_playwright.py]: playwright isn't used on it's own so the url has already been validated

File: autogpt/url_utils/validators.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        "https://127.0.0.1",
+        "https://0.0.0.0/",
+        "https://0.0.0.0",
+        "http://0.0.0.0/",
+        "http://0.0.0.0",
+        "http://0000",
+        "http://0000/",
+        "https://0000",
+        "https://0000/",
+    ]
```
Comment (ntindle) [inline autogpt/url_utils/validators.py:100]: In a separate PR may want to drop everything that has a scheme that's not in [`http`, `https`, `ws`, `git`].

See the list here https://en.wikipedia.org/wiki/List_of_URI_schemes

Comment (edcohen08) [reply autogpt/url_utils/validators.py:100]: agree with that

**Comment by Pwuts** [comment]
This will probably fix a lot of browsing issues, as the chrome driver will exit with really ugly errors even if the URL doesn't contain a schema.

File: autogpt/commands/web_playwright.py (hunk 1 above)
Comment (Pwuts) [reply autogpt/commands/web_playwright.py]: Does `validate_url` need to be imported though, if it is not used in the file?

Comment (edcohen08) [reply autogpt/commands/web_playwright.py]: Oh I misunderstood, definitely not it can be removed @Pwuts

**Review by Pwuts — APPROVED** [review]

**Pull request merged**